### PR TITLE
Something modifications to guidebook

### PR DESCRIPTION
**Pull request by Jueast**
Well,
It cannot merge automatically, please check it.

### CODE REVIEW

**Comment by sighingnow** [comment]
HI Jueast, thanks for your contribution. Please rebase the change locally and upload a replacement commit.

**Comment by sighingnow** [comment]
cc @fzyz999 @SivilTaram in case of you don't get notification.

**Review by sighingnow — CHANGES_REQUESTED** [review]
Please rebase to our master branch and resolve conflicts locally.

File: guide-book/chapters/4-syscall.tex [hunk 1]
```diff
@@ -160,11 +160,24 @@ \subsection{系统调用机制的实现}
 \subsection{基础系统调用函数}
 
 在系统调用机制搞定之后，我们自然是要弄几个系统调用玩一玩了。我们实现些什么系统调用呢？打开 lib/syscall\_all.c，可以看到玲琅满目的系统调用函数等着我们去填写。这些系统调用都是基础的系统调用，不论是之后的IPC还是fork，都需要这些基础的系统调用作为支撑。
-
-首先我们看向sys\_mem\_alloc。
-
-[Ping... Unreachabl]
-
+首先我们看向sys\_mem\_alloc。这个函数的主要功能是分配内存，简单的说来，用户程序可以通过这个系统调用给该程序所允许的虚拟内存空间内存分配实际的物理内存，需要用到一些我们之前在pmap.c中所定义的函数
```
Comment (sighingnow) [inline guide-book/chapters/4-syscall.tex]: Should be "简单来说" rather than "简单的说来".

File: guide-book/chapters/4-syscall.tex [hunk 2]
```diff
@@ -160,11 +160,24 @@ \subsection{系统调用机制的实现}
...
+\begin{exercise}
+实现lib/syscall\_all.c中的int sys\_mem\_alloc(int sysno,u\_int envid, u\_int va, u\_int perm)函数
+\end{exercise}
+我们再来看sys\_mem\_map,这个函数的参数很多，但是意义也很直接：将源进程地址空间中的相应内存映射到目标进程的相应地址空间的相应虚拟内存中去。换句话说，此时两者共享着一页物理内存。
+\begin{exercise}
+实现lib/syscall\_all.c中的int sys\_mem\_map(int sysno,u\_int srcid, u\_int srcva, u\_int dstid, u\_dstva, u\_int perm)函数
+\end{exercise}
+关于内存的还有一个函数：sys\_mem\_unmap,正如字面意义所显示的，这个系统调用的功能是解除某个进程地址空间虚拟内存和物理内存之间的映射关系。
```
Comment (sighingnow) [inline guide-book/chapters/4-syscall.tex]: Please use full width comma in ordinary text (after the sys\_mem\_unmap).

**Review by sighingnow — CHANGES_REQUESTED** [review]

**Comment by Jueast** [comment]
I have solve the conflicts manually without rebasing.

**Review by sighingnow — APPROVED** [review]

**Comment by sighingnow** [comment]
Thanks, @Jueast ,merged now.

**Comment by SivilTaram** [comment]
Sorry for being late. I will take a loot at it 👍

**Pull request merged**